app/controllers/response_templates_controller.rb

```diff
@@ -1,5 +1,29 @@
 class ResponseTemplatesController < ApplicationController
   after_action :verify_authorized
+  before_action :authenticate_user!, :ensure_json_request, only: %i[index]
+  rescue_from ArgumentError, with: :error_unprocessable_entity
+
+  MOD_TYPES = %w[mod_comment tag_adjustment].freeze
+  ADMIN_TYPES = %w[email_reply abuse_report_email_reply].freeze
+
+  def index
+    raise ArgumentError, "Missing param type_of" if params[:type_of].blank?
+
+    user_id = params[:type_of].include?("personal") ? current_user.id : nil
+    @response_templates = ResponseTemplate.where(type_of: params[:type_of], user_id: user_id)
```

Comment:
Might eventually want a composite index on these two fields.

Reply:
Good point, I'll add that now.